### PR TITLE
ENH: Add CI.

### LINKED ISSUE #23: Add CI
Add Continuous Integration to the module.

### PR DESCRIPTION
**Pull request by jhlegarreta**
- Add `*.yml` configuration files for CI.
- Add Python wrapping files.
- Update the `README.rst` file to reflect the CI status and Python
  installation steps.
- Modify the current `CTestConfig.cmake` file to make the build results
  be sent to the **Insight** project in **open.cdash.org**.

Resolves #23.

### CODE REVIEW

**Comment by jhlegarreta** [comment]
A few comments:
- The CI's will need to be enabled @phcerdan @thewtex
- The CI builds may trigger errors. But I'd be for merging this, and then gradually fixing the issues. This would allow us to be confident on our changes, and be a reminder of pending tasks.
- The wrapping is sure not to be accurate; I certainly need some help and clarifications about it.

File: README.rst [hunk 1]
```diff
@@ -1,6 +1,22 @@
...
 ==========================
 
+.. image:: https://circleci.com/gh/InsightSoftwareConsortium/ITKVariationalRegistration.svg?style=shield
+    :target: https://circleci.com/gh/InsightSoftwareConsortium/ITKVariationalRegistration
+
+.. image:: https://travis-ci.org/InsightSoftwareConsortium/ITKVariationalRegistration.svg?branch=master
+    :target: https://travis-ci.org/InsightSoftwareConsortium/ITKVariationalRegistration
+
+.. image:: https://img.shields.io/appveyor/ci/InsightSoftwareConsortium/itkvariationalregistration.svg
+    :target: https://ci.appveyor.com/project/InsightSoftwareConsortium/itkvariationalregistration
```
Comment (thewtex) [inline README.rst]: This should point to

https://ci.appveyor.com/project/InsightSoftwareConsortium/itkvariationalregistration

**Review by thewtex — CHANGES_REQUESTED** [review]
@jhlegarreta awesome!!!

CI-build enabled.

One minor issue noted inline.

File: README.rst (hunk 1 above)
Comment (jhlegarreta) [reply README.rst]: I do not see the typo :grimacing:

Comment (thewtex) [reply README.rst]: Sorry, pasted the wrong content.

```
    :target: https://ci.appveyor.com/project/InsightSoftwareConsortium/itkvariationalregistration
```

to

```
    :target: https://ci.appveyor.com/project/itkrobot/itkvariationalregistration
```

Comment (jhlegarreta) [reply README.rst]: Awesome. Thanks. Done in 8af03e9.

**Review by thewtex — APPROVED** [review]

**Comment by thewtex** [comment]
@dzenanz hats off to @jhlegarreta :tophat: , we now have CI testing for this repository. `master` could be updated for ITK 5 to help test #27

**Pull request merged**